operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -402,6 +410,27 @@ func (r *ReconcileIstioOperator) Reconcile(_ context.Context, request reconcile.
 	return reconcile.Result{}, err
 }
 
+func processDefaultWebhookAfterReconcile(iop *iopv1alpha1.IstioOperator, client kube.Client, exists bool) error {
+	var ns string
+	if configuredNamespace := iopv1alpha1.Namespace(iop.Spec); configuredNamespace != "" {
+		ns = configuredNamespace
+	} else {
+		ns = constants.IstioSystemNamespace
+	}
+	ignorePruneLabel := map[string]string{
+		helmreconciler.OwningResourceNotPruned: "true",
```

Comment:
why don't we prune this one? seems like this logic only applies to in-cluster operator as well - is that expected? 

since part of the goal was to make them consistent, we should probably comment in the code why we are inconsistent here?

Reply:
> why don't we prune this one

Other installed manifests are not pruned since here the manifests are being excluded for pruning. https://github.com/istio/istio/blob/9b7c62b698cd2ac8a7db4e5578e7c820265e3ac2/operator/pkg/helmreconciler/prune.go#L125

https://github.com/istio/istio/blob/9b7c62b698cd2ac8a7db4e5578e7c820265e3ac2/operator/pkg/helmreconciler/prune.go#L426

However the generated webhooks are not in the manifest list, as a result, if operator is doing a new reconcilation cycle, it will perform like `deleting generated webhooks -> watch a deletion event and trigger new iop reconcile -> install generated webhooks again -> deleting generated webhooks...` forever. 

> seems like this logic only applies to in-cluster operator as well - is that expected

This is expected, istioctl uninstall or IOP CR deletion are same as before, this is only to prevent webhooks being pruned during new reconcilation cycles like other installed resources.

Will add some comments.